langchain/util/sql_utils.ts

```diff
@@ -0,0 +1,116 @@
+import { SqlTable } from '../sql_database';
+import { DataSource } from 'typeorm';
+
+interface RawResultTableAndColumn {
+    table_name: string;
+    column_name: string;
+    data_type: string|undefined;
+    is_nullable: string;
+}
+
+export const verifyListTablesExistInDatabase = (tablesFromDatabase: Array<SqlTable>, listTables: Array<string>, errorPrefixMsg: string): void => {
+    const onlyTableNames: Array<string> = tablesFromDatabase.map((table: SqlTable) => table.tableName);
+    if (listTables.length > 0) {
+        for (const tableName of listTables) {
+            if (!onlyTableNames.includes(tableName)) {
+                throw new Error(
+                    `${errorPrefixMsg} the table ${tableName} was not found in the database`
+                );
+            }
+        }
+    }
+};
+
+export const verifyIncludeTablesExistInDatabase = (tablesFromDatabase: Array<SqlTable>, includeTables: Array<string>): void => {
+    verifyListTablesExistInDatabase(tablesFromDatabase, includeTables, 'Wrong include table name:');
+};
+
+export const verifyIgnoreTablesExistInDatabase = (tablesFromDatabase: Array<SqlTable>, ignoreTables: Array<string>): void => {
+    verifyListTablesExistInDatabase(tablesFromDatabase, ignoreTables, 'Wrong ignore table name:');
+};
+
+
+const formatToSqlTable = (rawResultsTableAndColumn: Array<RawResultTableAndColumn>): Array<SqlTable> => {
+    const sqlTable: Array<SqlTable> = [];
+    for (const oneResult of rawResultsTableAndColumn) {
+        const sqlColumn = {
+            columnName: oneResult.column_name,
+            dataType: oneResult.data_type,
+            isNullable: oneResult.is_nullable === 'YES',
+        };
+        const currentTable = sqlTable.find((oneTable) => oneTable.tableName === oneResult.table_name);
+        if (currentTable) {
+            currentTable.columns.push(sqlColumn);
+        } else {
+            const newTable = {
+                tableName: oneResult.table_name,
+                columns: [sqlColumn]
+            };
+            sqlTable.push(newTable);
+        }
+    }
+
+    return sqlTable;
+};
+
+
+export const getTableAndColumnsName = async (appDataSource: DataSource): Promise<Array<SqlTable>> => {
+    let sql;
+    if (appDataSource.options.type === 'postgres') {
```

Comment:
Followed Sean advice to get inspired by https://github.com/prisma/prisma-engines/blob/32d9b4f057e7645fdb9def515fb65ba881b65570/libs/sql-schema-describer/src/postgres/tables_query.sql